presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -115,10 +115,9 @@
 public class SemiTransactionalHiveMetastore
 {
     private static final Logger log = Logger.get(SemiTransactionalHiveMetastore.class);
-    private static final int PARTITION_COMMIT_BATCH_SIZE = 8;
     private static final int MAX_LAST_DATA_COMMIT_TIME_ENTRY_PER_TABLE = 100;
     private static final int MAX_LAST_DATA_COMMIT_TIME_ENTRY_PER_TRANSACTION = 10_000;
-
+    private static final int PARTITION_COMMIT_BATCH_SIZE = 100;
```

Comment:
What is the impact of this on other metastores ? would it cause the request to spike and cause cascading issues ?

Reply:
Current Hive versions of metastore backend with RDBMS should be able to handle 10000+ partitions.. Glue has an upper bound on 100 partitions per batch. information is listed in this commit message.

Reply:
sure will add comment to note the batch size setting

Reply:
SemiTransactionalHiveMetastore is used for all Metastores including Meta's metastore and Hudi and other metastores. Increasing the limit from 8 to 100 is not that simple and might over whelm some of the metastores. 

There are more metastores than Glue and old HiveMetastores.

Reply:
This should at least go behind a config and default should be left to 8, or more careful analysis is required.

Reply:
Better solution is to change the API, so that the final callers can decide how much they want to batch instead of batching upstream.

Reply:
good point, given there are various underlying metastore systems, let's give a default value of 10 in general, and caller will decide the batch partition size for commit, e.g. glue will have 100 partitions per guarded by `BATCH_CREATE_PARTITION_MAX_PAGE_SIZE` and process up to 100 requests in parallel so that BATCH_PARTITION_COMMIT_TOTAL_SIZE is 10000.